services/authzone.c

```diff
@@ -5958,6 +5961,18 @@ xfr_probe_send_probe(struct auth_xfer* xfr, struct module_env* env,
 				zname, master->host);
 			return 0;
 		}
+		if (auth_name != NULL) {
+			if (addr.ss_family == AF_INET
+			&&  ntohs(((struct sockaddr_in *)&addr)->sin_port)
+		            == 853)
+				((struct sockaddr_in *)&addr)->sin_port
+					= htons(53);
+			else if (addr.ss_family == AF_INET6
+			&&  ntohs(((struct sockaddr_in6 *)&addr)->sin6_port)
+		            == 853)
+                        	((struct sockaddr_in6 *)&addr)->sin6_port
+					= htons(853);
```

Comment:
This was a mistake, change 853 to 853, should have been 53. Also modified to use port number specified in unbound.conf.
```suggestion
					= htons(env->port);
```